Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2
# Dark Portal Raiders

## Changelog:
- Random Dark Portal event:
  - Draenor Raiders: 
    - May happen for the holder of the Dark Portal.
    - A group of raiders appears, you can let them through or face them in combat. If you let them through, they will attack the weakest ruler in the Easter Kingdoms.
    - `event WCDRA.200` in the console to force the event.


## Developer changelog:
- Got rid of passed side variable in `add_scandi_adventurer_target_effect` since it was used only inside the effect.
- `valid_adventurer_target_trigger` works fine for everyone, not only Scandinavian now.
- Adapted `scandinavian_adventurer_target_title_priority_value` for other cultures.
- Only vrykul prefer attacking other vrykul and coastal provinces.
- Adapted `is_valid_target_for_western_scandi_adventurer_trigger`, `significant_north_germanic_presence_trigger`, `significant_north_germanic_presence_trigger`, got rid of default north-germanic culture. All got renamed.
- Split `scandinavian_adventurer_start_war_effect` into separate effects.

# Tasks:
- [x] Move event localization to separate file
- [x] Test vrykul adventurers
- [x] Stop spawn vrykul if they can't conquer anything
- [x] Find out why orcs can't declare wars. Maybe there's big cultural presence in region?
- [x] Random events:
  - [x] Raiders pass the Portal (arakkoa, ogres, orcs, Lightbound draenei). Like Vikings, they start a war for some regions.
- [x] Adapt this event
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/122004969-200eed80-cdc6-11eb-87e2-732d3c804d37.png)

</details>

- [x] Twice less raiders


# How to test:
@Warcraft-GoA-Development-Team/ck3-testers Test the event (you can force its happening, `event WCDRA.200`, or wait for it), tell your opinion.
You can find Adventurers by looking for the Adventurer trait in the Character Finder.

@Warcraft-GoA-Development-Team/english-proofreaders open the [Page](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/pull/411/files?file-filters%5B%5D=.yml), scroll to the bottom to the `.yml` files, check them, suggest changes. If you forgot, here's the [Suggestion Guidelines](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/wiki/How-to-be-Tester-(For-CK3)#reviewing-code).
